app/core/utils.coffee

```diff
@@ -598,15 +598,19 @@ createLevelNumberMap = (levels) ->
 
 findNextLevel = (levels, currentIndex, needsPractice) ->
```

Comment:
I used the find all powers to trace all of the calls of this function, and could only see this method being used here:
https://github.com/codecombat/codecombat/blob/17987c470ff4025ecbc796f695015739cb20145f/app/models/Classroom.coffee#L173

Interested in a bit more clarity around the impact of this change within this function. I am also concerned that a value of -1 isn't handled:

https://github.com/codecombat/codecombat/blob/17987c470ff4025ecbc796f695015739cb20145f/app/models/Classroom.coffee#L178

Reply:
I don't think -1 is a new case to handle for `findNextLevel`, because it will also return levels.length if there are no levels left.  Either one gives an undefined index on the caller side.